api/resolvers/index.js

```diff
@@ -1651,15 +1651,116 @@ const resolvers = {
       if (!currentEventMember?.isAdmin)
         throw new Error("You need to be event admin to allocate funds.");
 
-      await new Allocation({
-        organizationId: currentOrgMember.organizationId,
-        eventId: targetEventMember.eventId,
-        eventMemberId,
-        amount,
-      }).save();
+      if (type === "ADD") {
+        await new Allocation({
+          organizationId: currentOrgMember.organizationId,
+          eventId: targetEventMember.eventId,
+          eventMemberId,
+          amount,
+        }).save();
+      } else if (type === "SET") {
+        const [
+          { totalAllocations } = { totalAllocations: 0 },
+        ] = await Allocation.aggregate([
+          {
+            $match: {
+              eventMemberId: mongoose.Types.ObjectId(eventMemberId),
+            },
+          },
+          { $group: { _id: null, totalAllocations: { $sum: "$amount" } } },
+        ]);
+
+        const [
+          { totalContributions } = { totalContributions: 0 },
+        ] = await Contribution.aggregate([
+          {
+            $match: {
+              eventMemberId: mongoose.Types.ObjectId(eventMemberId),
+            },
+          },
+          { $group: { _id: null, totalContributions: { $sum: "$amount" } } },
+        ]);
+
+        const balance = totalAllocations - totalContributions;
+
+        const adjustment = amount - balance;
+
+        await new Allocation({
+          organizationId: currentOrgMember.organizationId,
+          eventId: targetEventMember.eventId,
+          eventMemberId,
+          amount: adjustment,
+        }).save();
+      }
```

Comment:
Could be nice with an `} else {` to catch if you send a bad type (or does the graphql enum maybe check that for you?)

Reply:
Yep, the graphql enum will take care of that